asmcomp/emitaux.ml

```diff
@@ -361,7 +361,11 @@ let is_cfi_enabled () =
 
 let cfi_startproc () =
   if is_cfi_enabled () then
-    emit_string "\t.cfi_startproc\n"
+  begin
+    emit_string "\t.cfi_startproc\n";
+    if Config.architecture = "amd64" then
+      emit_string "endbr64;";
```

Comment:
This is not the proper place to add the `endbr64`. `cfi_startproc`, as the name suggests, only takes care of `cfi` annotation. I'd rather do this in `fundecl` in `asmcomp/amd64/emit.mlp`.